src/loki/polygon_search.cc

```diff
@@ -94,12 +100,21 @@ std::string to_geojson(const std::unordered_set<vb::GraphId>& edge_ids, vb::Grap
 namespace valhalla {
 namespace loki {
 
+// This custon within is created because using the boost::within for edge_info
+// shape will give different result depending on the order of the point.
+bool IsWithin(vb::EdgeInfo edge_info, polygon_type polygon) {
+  // create temporary line from edge_info
+  point_type p1(edge_info.shape()[0].first, edge_info.shape()[0].second);
+  point_type p2(edge_info.shape()[1].first, edge_info.shape()[1].second);
+  linestring_t edge_line{p1, p2};
+  return bg::within(edge_line, polygon);
```

Comment:
this doesn't looks right.. you're basically only testing the very first segment of the whole edge (or the last if it's reverse). I don't think your problems would become any less when you use the already existing (geographic) geometry types, but who knows, maybe that resolves the problem you describe in the comment. 

`shape will give different result depending on the order of the point.`: can you elaborate what "order of the point" means?

Reply:
in any case I don't think you should only test the very first segment here which might be the first 50 cm of a whole 2 km edge..

Reply:
Yep, you are correct. It was my bad unit test (most of the edge only has one segment). Anyway, this part is no longer exists after I merge to edges_in_rings.